packages/pretty-format/src/plugins/HTMLElement.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+
+'use strict';
+
+import type {Colors, Indent, Options, Print, Plugin} from '../types.js';
+
+const escapeHTML = require('./lib/escapeHTML');
+const HTML_ELEMENT_REGEXP = /(HTML\w*?Element)/;
+const test = (maybeSet: any) => isHTMLElement;
+
+function isHTMLElement(val: any) {
+  const toStringed = val.toString();
+  return HTML_ELEMENT_REGEXP.test(toStringed) ||
+    (toStringed === '[object Object]' &&
+      val.constructor &&
+      val.constructor.name &&
+      HTML_ELEMENT_REGEXP.test(val.constructor.name));
+}
```

Comment:
This is potentially a hot place, please keep perf in mind: https://twitter.com/bmeurer/status/846951275480711168